### PR TITLE
Removed exception in batch SFV

### PR DESCRIPTION
**Pull request by Akvinikym**
### Description of the Change

Throwing exception, if batch is formed incorrectly, was a bad idea, so it was replaced with error and failure of the whole proposal - it's still an exceptional case, so we don't want such damaged proposal to be propagated further.

### Benefits

No exception

### Possible Drawbacks 

None

### CODE REVIEW

File: irohad/validation/impl/stateful_validator_impl.cpp [hunk 1]
```diff
@@ -186,17 +186,30 @@ namespace iroha {
                 return tx.reducedHash() == batch_end_hash;
               });
           if (batch_end_it == txs_end) {
-            // for peer review: adequate exception variants?
-            throw std::runtime_error("Batch is formed incorrectly");
+            // exceptional case, such batch should not have passed stateless
+            // validation, so fail the whole proposal
+            transactions_errors_log.push_back(validation::TransactionError{
```
Comment (l4l) [inline irohad/validation/impl/stateful_validator_impl.cpp]: emplace_back + std::make_pair?

File: irohad/validation/impl/stateful_validator_impl.cpp [hunk 2]
```diff
@@ -186,17 +186,30 @@ namespace iroha {
...
+                validation::CommandError{
+                    "batch stateful validation",
```
Comment (l4l) [inline irohad/validation/impl/stateful_validator_impl.cpp]: use boost::fmt please

File: irohad/validation/impl/stateful_validator_impl.cpp [hunk 3]
```diff
@@ -186,17 +186,30 @@ namespace iroha {
                 return tx.reducedHash() == batch_end_hash;
               });
           if (batch_end_it == txs_end) {
-            // for peer review: adequate exception variants?
-            throw std::runtime_error("Batch is formed incorrectly");
+            // exceptional case, such batch should not have passed stateless
```
Comment (l4l) [inline irohad/validation/impl/stateful_validator_impl.cpp]: might be also useful print smth at log

**Comment by Akvinikym** [comment]
Please, also pay attentional that I fixed the SFV test, which breaks in develop every second time. The change is tiny, so I decided to put it here

File: test/module/irohad/validation/stateful_validator_test.cpp [hunk 4]
```diff
@@ -109,21 +109,22 @@ class Validator : public testing::Test {
...
     std::vector<shared_model::proto::Transaction> txs;
+    auto current_time = iroha::time::now();
 
-    for (const auto &creator : creators) {
+    for (size_t i = 0; i < creators.size(); ++i) {
       auto tx = TestTransactionBuilder()
-                    .creatorAccountId(creator)
-                    .createdTime(iroha::time::now())
+                    .creatorAccountId(creators[i])
+                    .createdTime(current_time + i)
```
Comment (l4l) [inline test/module/irohad/validation/stateful_validator_test.cpp]: :+1:

**Pull request merged**